File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode newNode = new ListNode(0);
        ListNode headNode = head;
        newNode.next = head;
        ListNode newNodeHead = newNode;
        int size = 0;
        int counter = 0;

        while (headNode != null) {
            size++;
            headNode = headNode.next;
        }

        while (newNodeHead != null) {
            if (counter == size - n) {
                ListNode nextNode = newNodeHead.next != null ? newNodeHead.next.next : null;
                newNodeHead.next = nextNode;
                break;
            }
            counter++;
            newNodeHead = newNodeHead.next;
        }
        return newNode.next;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        res = slow = fast = ListNode()
        fast.next = head

        for i in range(n):
            fast = fast.next
        
        while fast and fast.next:
            fast = fast.next
            slow = slow.next
        
    
        slow.next = slow.next.next

        return res.next

